### DISCUSSION TITLE
修改引擎项目代码后VS STDIO怎样才能快速生成解决方案？

### DISCUSSION THREAD

**Discussion by Idealist-dreamer** [BoomingTech/Piccolo#451] [GAMES104]
下载小引擎并使用VS STDIO开发，由于我经常需要在程序中输出变量来进行学习，所以我需要频繁的运行程序。
每一次运行程序都很慢，于是我上网查找资料，勾选了"当项目过期时"从不生成，于是VS STDIO直接运行不用编译了。
但是当我更改代码想要输出时，我发现直接运行的是以前的版本，于是我点击-生成-生成解决方案，结果还是很慢。
我点击配置管理器，仅勾选了PiccoloRuntime，我修改的engine.cpp，但是修改无效。
请问要勾选配置管理器里的哪些选项？如何设置VS STDIO做到只编译修改的文件，如何运行呢？

**Comment by HuajiKojima** [comment]
有一个“预编译头”的概念你可以学习一下，可以节省大部分的编译时间花销，不过这种预编译头很有可能会大幅更改代码结构以及包含顺序。而且这种方法更多的用于项目中使用的第三方库（比如标准库、IMGUI等）的编译，就引擎自身源码来说不敢完全保证有哪一部分你绝对不会进行热更新。
就目前VS的编译机制来说，如果你更改了某个类的声明或者某个宏定义，同时这个声明或者宏定义被多个其他的类所引用或者继承（文件层面上也就是这个类的声明或宏定义所属的头文件被多个其他的头文件或源文件包含），那么这些类是要重新编译的，即便这些类没有进行任何更改，这些时间开销注定无法被节省。
